src/Swarm/Language/Typecheck.hs

```diff
@@ -527,6 +527,8 @@ inferConst c = case c of
   Concat -> [tyQ| text -> text -> text |]
   Chars -> [tyQ| text -> int |]
   Split -> [tyQ| int -> text -> (text * text) |]
+  Char -> [tyQ| int -> text -> int |]
+  MkText -> [tyQ| int -> (int -> int) -> text |]
```

Comment:
Hmm, I can not imagine actually using `mkText` - creating the whole text just based on length and indices sounds like it will require a rather complicated index-to-char function.

What I would really like to use though is the `singleton` so the special version:
```
\c. mkText 1 (\_.c)
```

I can see the theoretical appeal of the proposed functions, but I worry they will not be easy for players to use.

On the other hand, a `singleton` (or `toChar`) fills the space between `++` and `charAt`, so that you can roundtrip without outside references like ASCII or Unicode table.